beets/importer.py

```diff
@@ -1085,7 +1122,7 @@ def read_tasks(session):
 
             log.debug(u'extracting archive {0}',
                       displayable_path(toppath))
-            archive_task = ArchiveImportTask(toppath)
+            archive_task = task_factory.archive(toppath)[0]
             try:
                 archive_task.extract()
```

Comment:
This is now incorrect, right? `archive_task` here is at least a list, and may not even be a list of `ArchiveTask`s if a plugin produces something else.